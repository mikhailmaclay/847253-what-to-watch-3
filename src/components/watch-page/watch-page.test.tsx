// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// Components
import WatchPage from './watch-page';
//
import emptyArrowFunction from '../../utils/components/empty-arrow-function.js';

describe(`<WatchPage/>`, () => {
  const props = {
    id: 1,
    name: `Hello World`,
    backgrounds: [`black`, ``],
    video: ``,
    onClose: emptyArrowFunction
  };

  it(`should render correctly`, () => {
    const result = renderer.create(
        <WatchPage {...props}/>, {createNodeMock: () => ({})}
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
