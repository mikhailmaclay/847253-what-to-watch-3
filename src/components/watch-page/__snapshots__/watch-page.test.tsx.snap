// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchPage/> should render correctly 1`] = `
<div
  className="player"
  style={
    Object {
      "backgroundColor": "black",
      "userSelect": "none",
    }
  }
>
  <object
    height="100%"
    name="video"
    style={
      Object {
        "display": "flex",
      }
    }
    width="100%"
  >
    <video
      height="100%"
      name="Hello World"
      onCanPlayThrough={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
      onDurationChange={[Function]}
      onEnded={[Function]}
      onLoadedMetadata={[Function]}
      onMouseMove={[Function]}
      onPause={[Function]}
      onPlay={[Function]}
      onPlaying={[Function]}
      onTimeUpdate={[Function]}
      onWaiting={[Function]}
      poster=""
      src=""
      style={
        Object {
          "objectFit": "cover",
        }
      }
      width="100%"
    />
    <button
      className="player__exit"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      type="button"
    >
      Exit
    </button>
    <div
      className="player__controls"
      style={
        Object {
          "backgroundImage": "linear-gradient(transparent, black)",
          "zIndex": "999",
        }
      }
    >
      <div
        className="player__controls-row"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <title>
            Loading...
          </title>
          <use
            xlinkHref="#activity-indicator"
          />
        </svg>
        <div
          className="player__name"
        >
          Hello World
        </div>
      </div>
    </div>
  </object>
</div>
`;
